src/main/java/edu/cornell/kfs/module/purap/document/authorization/IWantDocumentPresentationController.java

```diff
@@ -81,6 +92,48 @@ public boolean canEditDocumentOverview(Document document) {
                         workflowDocument, GlobalVariables.getUserSession().getPrincipalId());
     }
 
+    /* 
+     * CU Customization:
+     * Everyone should be able to view the Contract tab when the document is Final.
+     */
+    public boolean canViewContractTab(Document document) {
+        WorkflowDocument workflowDocument = document.getDocumentHeader().getWorkflowDocument();
+        return isInOrgHierarchyOrPurchasingAssistantNode(document) || workflowDocument.isFinal();
+    }
+
+    /*
+     * CU Customization:
+     * We restrict the editing of the Contract Tab contents on IWNT docs to enroute status at the
+     * OrganizationHierarchy node and PurchasingContractAssistant node.
+     */
+    public boolean canEditContractIndicator(Document document) {
+        WorkflowDocument workflowDocument = document.getDocumentHeader().getWorkflowDocument();
+        boolean isInApproversNode = isInApproversNode(document);
+        
+        return workflowDocument.isEnroute()
+                && isInOrgHierarchyOrPurchasingAssistantNode(document)
+                && isInApproversNode;
+    }
+    
+    private boolean isInApproversNode(Document document) {
+        final Person currentUser = GlobalVariables.getUserSession().getPerson();
```

Comment:
Good question. I am not seeing the Authorizer getEditModes being used in this fashion. That seems to remove editModes when users are not Authorized. Should the edit modes be added  by default in the presentation controller and then removed in the authorizer getEditModes if the user does not have access?

Reply:
Yes, I think it would be better if the non-Person-specific checks (in the presentation controller) add the edit mode as needed, and then the Person-specific checks (in the authorizer) remove the edit mode as needed. The presentation controller is typically for performing checks that do NOT depend on who the current user is, whereas the authorizer is typically for performing checks that DO depend on who the current user is.

Reply:
updated